Mathlib/Analysis/NormedSpace/Unitization.lean

```diff
@@ -0,0 +1,265 @@
+`Unitization.splitMul 𝕜 A`. When `A` is a `RegularNormedAlgebra`, then
+`ContinuousLinearMap.mul 𝕜 A` is an isometry, and is therefore automatically injective. -/
+theorem splitMul_injective_of_clm_mul_injective
+    (h : Function.Injective (mul 𝕜 A)) :
+    Function.Injective (splitMul 𝕜 A) := by
+  rw [injective_iff_map_eq_zero]
+  intro x hx
+  induction x using Unitization.ind
+  rw [map_add] at hx
+  simp only [Prod.mk_add_mk, add_zero, fst_inl, splitMul_apply,
+    snd_inl, snd_inr, Prod.mk_eq_zero, zero_add, fst_inr,
+    map_zero, splitMul_apply, add_zero, Prod.mk_eq_zero] at hx
+  obtain ⟨rfl, hx⟩ := hx
+  simp only [map_zero, zero_add, inl_zero] at hx ⊢
+  rw [← map_zero (mul 𝕜 A)] at hx
+  rw [h hx, inr_zero]
+
+variable [RegularNormedAlgebra 𝕜 A]
+variable (𝕜 A)
+
+/- In a `RegularNormedAlgebra`, the map `Unitization.splitMul 𝕜 A` is injective. We will use this
+to pull back the norm from `𝕜 × (A →L[𝕜] A)` to `Unitization 𝕜 A`. -/
+theorem splitMul_injective : Function.Injective (splitMul 𝕜 A) :=
+  splitMul_injective_of_clm_mul_injective (isometry_mul 𝕜 A).injective
+
+variable {𝕜 A}
+
+section Aux
+
+/-- Pull back the normed ring structure from `𝕜 × (A →L[𝕜] A)` to `Unitization 𝕜 A` using the
+algebra homomorphism `Unitization.splitMul 𝕜 A`. This does not give us the desired topology,
+uniformity or bornology on `Unitization 𝕜 A` (which we want to agree with `Prod`), so we only use
+it as a local instance to build the real one. -/
+@[reducible]
+noncomputable def normedRingAux : NormedRing (Unitization 𝕜 A) :=
+  @NormedRing.induced _ (Unitization 𝕜 A) (𝕜 × (A →L[𝕜] A)) Unitization.instRing
+    Prod.normedRing _ (splitMul 𝕜 A) (splitMul_injective 𝕜 A)
+-- ummmm... what? why does Lean need me to fill in these instances?
+
+attribute [local instance] Unitization.normedRingAux
+
+/-- Pull back the normed algebra structure from `𝕜 × (A →L[𝕜] A)` to `Unitization 𝕜 A` using the
+algebra homomorphism `Unitization.splitMul 𝕜 A`. This uses the wrong `NormedRing` instance (i.e.,
+`Unitization.normedRingAux`), so we only use it as a local instance to build the real one.-/
+@[reducible]
+noncomputable def normedAlgebraAux : NormedAlgebra 𝕜 (Unitization 𝕜 A) :=
+  NormedAlgebra.induced 𝕜 (Unitization 𝕜 A) (𝕜 × (A →L[𝕜] A)) (splitMul 𝕜 A)
+
+attribute [local instance] Unitization.normedAlgebraAux
+
+theorem norm_def (x : Unitization 𝕜 A) : ‖x‖ = ‖splitMul 𝕜 A x‖ :=
+  rfl
+
+/-- This is often the more useful lemma to rewrite the norm as opposed to `Unitization.norm_def`. -/
+theorem norm_eq_sup (x : Unitization 𝕜 A) :
+    ‖x‖ = ‖x.fst‖ ⊔ ‖algebraMap 𝕜 (A →L[𝕜] A) x.fst + mul 𝕜 A x.snd‖ := by
+  rw [norm_def, splitMul_apply, Prod.norm_def, sup_eq_max]
+
+variable (𝕜 A)
+
+/-- The identity map between `Unitization 𝕜 A` and `𝕜 × A` as an `AddEquiv`. -/
+protected def addEquiv : Unitization 𝕜 A ≃+ 𝕜 × A :=
+  AddEquiv.refl _
+
+variable {𝕜 A}
+
+theorem lipschitzWith_addEquiv :
+    LipschitzWith (2 : ℝ).toNNReal (Unitization.addEquiv 𝕜 A) := by
```

Comment:
After some offline discussion, we came to the conclusion that this approach doesn't really avoid the need for the local instances because the argument that the bornologies coincide still uses the metric structure.